include/onnxruntime/core/session/onnxruntime_run_options_config_keys.h

```diff
@@ -42,3 +42,8 @@ static const char* const kOrtRunOptionsConfigQnnPerfModePostRun = "qnn.htp_perf_
 
 // Set RPC control latency for QNN HTP backend
 static const char* const kOrtRunOptionsConfigQnnRpcControlLatency = "qnn.rpc_control_latency";
+
+// Set graph annotation id for CUDA EP. Use with enable_cuda_graph=true.
+// The value should be an integer. If the value is not set, ORT session only captures one cuda graph.
+// If the value is set to -1, cuda graph capture/replay is disabled in that run.
+static const char* const kOrtRunOptionsConfigCudaGraphAnnotation = "ep.cuda.cuda_graph_annotation";
```

Comment:
How about rename as cuda_graph_id

Reply:
Maybe keep it generic  (i.e.) without tying it to "cuda" in the RunOption? That may allow other EPs to re-use it if needed.

Reply:
Good suggestion. names like `capture_graph_id` or `gpu_graph_id` might be used in CUDA/TRT/ROCm.

Reply:
sounds good